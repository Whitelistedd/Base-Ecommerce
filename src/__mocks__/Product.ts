export const TEST_PRODUCT1 = {
	categories: ['hoodies'],
	color: ['black'],
	createdAt: '2022-06-16T11:18:10.448Z',
	desc: '',
	gender: ['men'],
	img: [
		'https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp',
	],
	inStock: 8,
	price: 3000,
	size: ['XL', 'L', 'M'],
	title: "MEN'S QUARTER ZIP HOODIE",
	updatedAt: '2022-06-16T11:18:10.448Z',
	__v: 0,
	_id: '62ab1172bb9ddb4cfa932c50',
}
export const TEST_PRODUCT2 = {
	categories: ['hoodies'],
	color: ['black'],
	createdAt: '2022-06-16T11:18:10.448Z',
	desc: '',
	gender: ['men'],
	img: [
		'https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp',
	],
	inStock: 8,
	price: 3000,
	size: ['XL', 'L', 'M'],
	title: "MEN'S QUARTER ZIP HOODIE",
	updatedAt: '2022-06-16T11:18:10.448Z',
	__v: 0,
	_id: '62ab1172bb9ddb4cfa932c51',
}
export const TEST_PRODUCT3 = {
	categories: ['hoodies'],
	color: ['black'],
	createdAt: '2022-06-16T11:18:10.448Z',
	desc: '',
	gender: ['men'],
	img: [
		'https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp',
	],
	inStock: 8,
	price: 3000,
	size: ['XL', 'L', 'M'],
	title: "MEN'S QUARTER ZIP HOODIE",
	updatedAt: '2022-06-16T11:18:10.448Z',
	__v: 0,
	_id: '62ab1172bb9ddb4cfa932c52',
}
