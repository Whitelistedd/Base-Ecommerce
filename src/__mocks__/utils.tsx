/* eslint-disable react/display-name */

import * as React from 'react'

import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

import { render } from '@testing-library/react'

const createTestQueryClient = () =>
	new QueryClient({
		defaultOptions: {
			queries: {
				retry: false,
			},
		},
		logger: {
			log: console.log,
			warn: console.warn,
			// eslint-disable-next-line @typescript-eslint/no-empty-function
			error: () => {},
		},
	})

export function renderWithClient(client: QueryClient, ui: React.ReactElement) {
	const { rerender, ...result } = render(
		<QueryClientProvider client={client}>{ui}</QueryClientProvider>
	)
	return {
		...result,
		rerender: (rerenderUi: React.ReactElement) =>
			rerender(
				<QueryClientProvider client={client}>{rerenderUi}</QueryClientProvider>
			),
	}
}

export function createWrapper() {
	const testQueryClient = createTestQueryClient()
	return ({ children }: { children: React.ReactNode }) => (
		<QueryClientProvider client={testQueryClient}>
			{children}
		</QueryClientProvider>
	)
}
