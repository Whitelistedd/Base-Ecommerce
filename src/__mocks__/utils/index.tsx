import { Provider } from 'react-redux'
import { QueryClientProvider } from '@tanstack/react-query'
import { UserProvider } from '@auth0/nextjs-auth0'
import { createTestQueryClient } from './reactQuery'
import { mockStore } from './reactRedux'

export function wrapWithAll(children: React.ReactNode) {
  const testQueryClient = createTestQueryClient()
  return (
    <Provider store={mockStore}>
      <QueryClientProvider client={testQueryClient}>
        <UserProvider>{children}</UserProvider>
      </QueryClientProvider>
    </Provider>
  )
}
