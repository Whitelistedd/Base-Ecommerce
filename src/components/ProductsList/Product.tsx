import React, { useState } from 'react'
import Link from 'next/link'
import styled, { keyframes } from 'styled-components'
import Image from 'next/image'

import { devices } from '../../data'
import { ProductProps } from './ProductsList.model'

export const Product: React.FC<ProductProps> = ({ item }) => {
  const [hoveredStatus, setHoveredStatus] = useState(false)
  return (
    <Container
      className="Product"
      onMouseEnter={() => setHoveredStatus(true)}
      onMouseLeave={() => setHoveredStatus(false)}
    >
      <StyledLink images={item.img} href={`/product/${item._id}`}>
        <InfoContainer
          className="ProductInfo"
          hoveredStatus={hoveredStatus}
          inStock={item.inStock}
        >
          <Image1 className="ProductImage">
            {/* если пользователь наводит курсор на изображение, он покажет второе изображение продукта, а если нет, то покажет первое */}
            <StyledImage
              className="Image1"
              src={item.img[0]}
              layout="responsive"
              width={300}
              height={400}
              alt={item.title}
            />
          </Image1>
          <Image2 className="ProductImage">
            <StyledImage
              className="Image2"
              src={item.img[1]}
              layout="responsive"
              width={300}
              height={400}
            />
          </Image2>
          <Title>{item.title}</Title>
          {/* если товар распродан, появится это сообщение */}
          {!item.inStock && <SoldOut>Распродан</SoldOut>}
        </InfoContainer>
      </StyledLink>
    </Container>
  )
}

const Title = styled.p`
  font-size: 1em;
  font-style: normal;
  font-weight: 700;
  letter-spacing: 1px;
  text-align: center;
`

const SoldOut = styled(Title)`
  color: #ff0000;
`

const unFade = keyframes`
 0% {
  opacity: 0;
 }
 100% {
  opacity: 1;
 }
`

const Image1 = styled.div`
  animation: 200ms ${unFade} linear;
`

const Image2 = styled(Image1)`
  display: none;
`

const StyledImage = styled(Image)``

const InfoContainer = styled.div<{ inStock: boolean; hoveredStatus: boolean }>`
  min-width: 300px;
  max-width: 300px;
  height: 100%;
  width: 100%;
  object-fit: cover;

  &:hover {
    ${Image1} {
      display: none;
    }
    ${Image2} {
      display: block;
    }
    cursor: pointer;
  }

  span {
    ${(props) =>
      !props.inStock
        ? `
  filter: grayscale(0.5) brightness(60%); 
  `
        : ``}
  }
`

const StyledLink = styled(Link)<{ images: Array<string> }>``

const Container = styled.div`
  color: black;
  display: flex;
  gap: 0.5em;
  align-items: center;
  justify-content: center;
  position: relative;
  flex-direction: column;
  text-decoration: none;
  padding: 1em;
  box-shadow: 0px 0px 0px black;
  border-radius: 10px;
  height: 100%;

  @media only screen and (max-width: ${devices.Phone}px) {
    min-width: 300px;
  }
`
