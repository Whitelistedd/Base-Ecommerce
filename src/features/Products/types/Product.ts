import { ProductDataType } from 'GlobalTypes/GlobalTypes.model'

export interface ProductProps {
  item: ProductDataType
  className?: string
}
