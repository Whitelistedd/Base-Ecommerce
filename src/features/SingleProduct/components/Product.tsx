import { Container, ProductsWrap } from '../assets/Product-styles'

import { ImageSwipe } from './Images/ImageSwipe'
import { ProductForm } from './ProductForm'
import { ProductImages } from './Images/ProductImages'
import { ProductProps } from '../types/SingleProduct.model'
import React from 'react'

export const Product: React.FC<ProductProps> = ({ product }) => {
  return (
    <Container>
      <ProductsWrap>
        <ImageSwipe productInfo={product} />
        <ProductImages productInfo={product} />
        <ProductForm productInfo={product} />
      </ProductsWrap>
    </Container>
  )
}
