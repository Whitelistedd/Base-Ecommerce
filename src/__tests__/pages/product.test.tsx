import { QueryCache, QueryClient } from '@tanstack/react-query'
import {
  renderWithClient,
  wrapWithClient,
} from '../../__mocks__/utils/reactQuery'

import SingleProductPage from '../../pages/product/[id]'
import { TEST_PRODUCT1 } from '@/__mocks__/Product'
import renderer from 'react-test-renderer'
import { useProduct } from '@/features/SingleProduct'
import { wrapWithAll } from '@/__mocks__/utils'

jest.mock('next/router', () => require('next-router-mock'))

const mockedUseProduct = useProduct as jest.Mock

jest.mock('@/features/SingleProduct', () => ({
  useProduct: jest.fn(),
}))

describe('Index page', () => {
  const queryCache = new QueryCache()
  const queryClient = new QueryClient({ queryCache })
  const mockProductData = TEST_PRODUCT1
  it('fetches products with correct params', () => {
    mockedUseProduct.mockImplementation(() => ({
      status: 'success',
      data: mockProductData,
    }))
    renderWithClient(
      queryClient,
      wrapWithAll(<SingleProductPage product={mockProductData} />)
    )

    expect(useProduct).toHaveBeenCalledWith('0', mockProductData)
  })

  describe('while loading', () => {
    it('renders a loader', () => {
      mockedUseProduct.mockImplementation(() => ({
        status: 'loading',
        data: mockProductData,
      }))

      const { getByAltText } = renderWithClient(
        queryClient,
        wrapWithAll(<SingleProductPage product={mockProductData} />)
      )

      expect(getByAltText(/loading.../)).toBeInTheDocument()
    })
  })

  describe('with an error', () => {
    it('renders a error', () => {
      mockedUseProduct.mockImplementation(() => ({
        status: 'error',
        data: mockProductData,
      }))

      const { getByAltText } = renderWithClient(
        queryClient,
        wrapWithAll(<SingleProductPage product={mockProductData} />)
      )

      expect(getByAltText(/Failed/)).toBeInTheDocument()
    })
  })

  it('should match snapshot', () => {
    mockedUseProduct.mockImplementation(() => ({
      status: 'success',
      data: mockProductData,
    }))
    const tree = renderer
      .create(wrapWithAll(<SingleProductPage product={mockProductData} />))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
