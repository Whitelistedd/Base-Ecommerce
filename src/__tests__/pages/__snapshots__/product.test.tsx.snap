// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should match snapshot 1`] = `
<div
  className="sc-122fb789-1 kmREFj"
>
  <div
    className="sc-122fb789-0 dsWDdN"
  >
    <div
      className="swiper-container sc-8e5a2b4-1 jiheNC"
      modules={
        [
          {
            "create": [Function],
            "name": "navigation",
            "on": {
              "click": [Function],
              "destroy": [Function],
              "enable disable": [Function],
              "fromEdge": [Function],
              "init": [Function],
              "toEdge": [Function],
            },
            "params": {
              "navigation": {
                "disabledClass": "swiper-button-disabled",
                "hiddenClass": "swiper-button-hidden",
                "hideOnClick": false,
                "lockClass": "swiper-button-lock",
                "nextEl": null,
                "prevEl": null,
              },
            },
          },
          {
            "create": [Function],
            "name": "effect-fade",
            "on": {
              "beforeInit": [Function],
              "setTransition": [Function],
              "setTranslate": [Function],
            },
            "params": {
              "fadeEffect": {
                "crossFade": false,
              },
            },
          },
        ]
      }
    >
      <div
        className="swiper-button-prev"
      />
      <div
        className="swiper-button-next"
      />
      <div
        className="swiper-wrapper"
      >
        <div
          className="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index={0}
        >
          <div
            className="sc-4c7a1ac9-1 sc-b302848f-0 fcXFCP bwMZUN  sc-8e5a2b4-0 gfdRvo"
            onClick={[Function]}
          >
            <span
              style={
                {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "opacity": 1,
                    "padding": 0,
                    "paddingTop": "132.6086956521739%",
                    "width": "initial",
                  }
                }
              />
              <img
                alt="ProductImage "
                className="sc-4c7a1ac9-0"
                data-nimg="responsive"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="ProductImage "
                  className="sc-4c7a1ac9-0"
                  data-nimg="responsive"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=3840&q=75"
                  srcSet="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=3840&q=75 3840w"
                  style={
                    {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </div>
        </div>
        <div
          className="swiper-slide"
          data-swiper-slide-index={0}
        >
          <div
            className="sc-4c7a1ac9-1 sc-b302848f-0 fcXFCP bwMZUN  sc-8e5a2b4-0 gfdRvo"
            onClick={[Function]}
          >
            <span
              style={
                {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "opacity": 1,
                    "padding": 0,
                    "paddingTop": "132.6086956521739%",
                    "width": "initial",
                  }
                }
              />
              <img
                alt="ProductImage "
                className="sc-4c7a1ac9-0"
                data-nimg="responsive"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="ProductImage "
                  className="sc-4c7a1ac9-0"
                  data-nimg="responsive"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=3840&q=75"
                  srcSet="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=3840&q=75 3840w"
                  style={
                    {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </div>
        </div>
        <div
          className="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index={0}
        >
          <div
            className="sc-4c7a1ac9-1 sc-b302848f-0 fcXFCP bwMZUN  sc-8e5a2b4-0 gfdRvo"
            onClick={[Function]}
          >
            <span
              style={
                {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "opacity": 1,
                    "padding": 0,
                    "paddingTop": "132.6086956521739%",
                    "width": "initial",
                  }
                }
              />
              <img
                alt="ProductImage "
                className="sc-4c7a1ac9-0"
                data-nimg="responsive"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="ProductImage "
                  className="sc-4c7a1ac9-0"
                  data-nimg="responsive"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=3840&q=75"
                  srcSet="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=3840&q=75 3840w"
                  style={
                    {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-21e3326b-2 bznTfU"
    >
      <div
        className="sc-21e3326b-1 kHOUgV"
      >
        <div
          className="sc-4c7a1ac9-1 sc-b302848f-0 fcXFCP bwMZUN active "
          onClick={[Function]}
        >
          <span
            style={
              {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "opacity": 1,
                  "padding": 0,
                  "paddingTop": "132.6086956521739%",
                  "width": "initial",
                }
              }
            />
            <img
              alt="ProductImage "
              className="sc-4c7a1ac9-0"
              data-nimg="responsive"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="ProductImage "
                className="sc-4c7a1ac9-0"
                data-nimg="responsive"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=3840&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F93ksW04%2FUntitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp&w=3840&q=75 3840w"
                style={
                  {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </div>
      </div>
      <img
        alt="MEN'S QUARTER ZIP HOODIE"
        className="sc-21e3326b-0 sc-3877b992-0 bZCIJJ dLeqKL"
        src="https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp"
      />
    </div>
    <div
      className="sc-7fdd42f4-11 iEufdy"
    >
      <h1
        className="sc-7fdd42f4-0 czuFGP"
      >
        MEN'S QUARTER ZIP HOODIE
      </h1>
      <span
        className="sc-7fdd42f4-1 fnalvN"
      >
        ₽
        3000
      </span>
      <div
        className="sc-8c48ebb0-0 jlXspZ"
      >
        <h4
          className="sc-8c48ebb0-3 jMuDkq"
        >
          Size :
        </h4>
        <div
          className="sc-8c48ebb0-1 juRuoY"
        >
          <label
            className="sc-d46022de-4 kNiYOK"
          >
            <input
              className="sc-d46022de-1"
              name="size"
              onChange={[Function]}
              type="radio"
              value="XL"
            />
            <span
              className="sc-d46022de-3 vihxs"
            >
              <div
                className="sc-d46022de-0"
              >
                <h3
                  className="sc-d46022de-2"
                >
                  XL
                </h3>
              </div>
            </span>
          </label>
          <label
            className="sc-d46022de-4 kNiYOK"
          >
            <input
              className="sc-d46022de-1"
              name="size"
              onChange={[Function]}
              type="radio"
              value="L"
            />
            <span
              className="sc-d46022de-3 vihxs"
            >
              <div
                className="sc-d46022de-0"
              >
                <h3
                  className="sc-d46022de-2"
                >
                  L
                </h3>
              </div>
            </span>
          </label>
          <label
            className="sc-d46022de-4 kNiYOK"
          >
            <input
              className="sc-d46022de-1"
              name="size"
              onChange={[Function]}
              type="radio"
              value="M"
            />
            <span
              className="sc-d46022de-3 vihxs"
            >
              <div
                className="sc-d46022de-0"
              >
                <h3
                  className="sc-d46022de-2"
                >
                  M
                </h3>
              </div>
            </span>
          </label>
        </div>
      </div>
      <div
        className="sc-8c48ebb0-0 jlXspZ"
      >
        <h4
          className="sc-8c48ebb0-3 jMuDkq"
        >
          Colors : 
          <span
            className="sc-8c48ebb0-2 fNvPJh"
          />
           
        </h4>
        <div
          className="sc-8c48ebb0-1 juRuoY"
        >
          <label
            className="sc-5fc067ee-4 bNICJU sc-8c48ebb0-4 gDScID"
          >
            <input
              className="sc-5fc067ee-1"
              name="color"
              onChange={[Function]}
              type="radio"
              value="black"
            />
            <span
              className="sc-5fc067ee-3 gsVWtG"
            >
              <div
                className="sc-5fc067ee-0"
              >
                <h3
                  className="sc-5fc067ee-2 jlZsxx"
                />
              </div>
            </span>
          </label>
        </div>
      </div>
      <div
        className="sc-7fdd42f4-8 iMpeaW"
      >
        <div
          className="sc-7fdd42f4-9 dPKUer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RemoveIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
          <span
            className="sc-7fdd42f4-10 cYgmPW"
          >
            1
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </div>
        <button
          className="sc-7fdd42f4-6 jLGeHu"
          onClick={[Function]}
        >
          ADD TO CART
        </button>
      </div>
      <div
        className="sc-7fdd42f4-2 iliLAj"
      >
        <div
          className="sc-7fdd42f4-3 kna-dXG"
        />
      </div>
    </div>
  </div>
</div>
`;
